addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -55,12 +55,139 @@ def set_crm_team_message_types(env):
             (4, env.ref('sale.mt_salesteam_invoice_created').id)]})
 
 
+@openupgrade.logging()
+def set_so_line_amounts(env):
+    """We replicate here the code of the compute function and set the values
+    finally via SQL for avoiding the trigger of the rest of the computed
+    fields that depends on these fields.
+    """
+    lines = env['sale.order.line'].search([])
+    for line in lines:
+        price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
+        taxes = line.tax_id.compute_all(
+            price, line.order_id.currency_id, line.product_uom_qty,
+            product=line.product_id, partner=line.order_id.partner_id)
+        env.cr.execute("""
+            UPDATE sale_order_line
+            SET price_tax = %s,
+                price_total = %s,
+                price_subtotal = %s
+            WHERE id = %s""", (
+                taxes['total_included'] - taxes['total_excluded'],
+                taxes['total_included'],
+                taxes['total_excluded'],
+                line.id,
+            )
+        )
+
+
+def set_so_line_computed_rest(env):
+    """Emulate the computation of the rest of the computed fields through
+    equivalente SQL querys.
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line
+        SET price_reduce = CASE
+            WHEN abs(product_uom_qty) > 0.00001
+                THEN price_subtotal / product_uom_qty
+            ELSE 0.0
+        END""")
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET qty_invoiced = sub.qty
+        FROM (
+            SELECT rel.order_line_id,
+                SUM(CASE
+                    WHEN ai.type = 'out_invoice'
+                        THEN ail.quantity / u.factor * u2.factor
+                    WHEN ai.type = 'out_refund'
+                        THEN -ail.quantity / u.factor * u2.factor
+                    ELSE 0.0
```

Comment:
```suggestion
                    WHEN ai.type IN ('out_invoice', 'out_refund') AND u.category_id != u2.category_id
                        THEN ail.quantity
                    ELSE 0.0
```

This suggestion is in combination with the 2 above, to match this code: https://github.com/OCA/OpenUpgrade/blob/bd1a1d7b04b1d5ea5d94b931443d61fc71d810a2/addons/product/product.py#L120-L124